components/backup-stream/src/endpoint.rs

```diff
@@ -99,9 +101,11 @@ where
             // spawn a worker to watch task changes from etcd periodically.
             let meta_client_clone = meta_client.clone();
             let scheduler_clone = scheduler.clone();
+            let pool_clone = pool.clone();
             // TODO build a error handle mechanism #error 2
             pool.spawn(async {
-                if let Err(err) = Self::starts_watch_tasks(meta_client_clone, scheduler_clone).await
+                if let Err(err) =
```

Comment:
can we remove the Arc of pool, and  move start_watch_task/starts_watch_pause here 🤔